### PR TITLE
Review the future dates used in test cases #4707

### LINKED ISSUE #4707: Review the future dates used in test cases
Continuation of #4701.
Need to review which future dates can be set to the past (so that things like #4701 can be prevented for good), and which ones need to be in the future (there is at least one such case in `typicalDataBundle`).
Similar bug to #4701 will haunt us in approximately 1-2 months from now if this is not fixed for good.

### PR DESCRIPTION
**Pull request by tarun14110**
Fixes #4707

### CODE REVIEW

**Comment by tarun14110** [comment]
@thyageshm @wkurniawan07  I changed tests to show dev green on 2nd March by shifting date to 1st march. Hardcoded for the time being. Issue is only partially solved .
Now this problem will arise again in next month . So we have some time for that to take work on.
I will soon update with complete solution.

**Comment by thyageshm** [comment]
@wkurniawan07 @tarun14110 I think we can open a second pr for the same issue to fix the rest of the dates. Or if it wouldnt take too long, can we fix it properly by the end of today in this pr?

@damithc what do you think?

**Comment by tarun14110** [comment]
@thyageshn I will send another PR by tonight. Actually, we will have failed cases for today only. So, if we are not meging  this  .we can work on that properly by tomorrow .

**Comment by damithc** [comment]
How many tests are affected?

**Comment by damithc** [comment]
Do  quick fix by changing the hardcoded values and then do a proper fix by next milestone?

**Comment by thyageshm** [comment]
@damithc I think in a few hours it wouldn't matter anymore... I guess the issue can be fixed by next milestone. We may not need a quick fix...

**Comment by damithc** [comment]
Sure.

**Comment by tarun14110** [comment]
@wkurniawan07 I have updated all hardcoded dates from 2016 to 2026, and from 2017 to 2027. Please have a look.

**Comment by damithc** [comment]
Please squash all related commits into one. Too many commits will make the commit tree unnecessarily messy.

**Comment by tarun14110** [comment]
done

File: src/test/resources/data/StudentFeedbackResultsPageUiTest.json [hunk 1]
```diff
@@ -1831,11 +1831,11 @@
             "feedbackQuestionId":"1",
             "giverEmail":"SFResultsUiT.instr@gmail.tmt",
             "feedbackResponseId":"1%SFResultsUiT.alice.b@gmail.tmt%SFResultsUiT.alice.b@gmail.tmt",
-            "createdAt":"2016-03-01 11:59 PM UTC",
+            "createdAt":"2026-03-01 11:59 PM UTC",
             "showCommentTo":[STUDENTS];
             "showGiverNameTo":[STUDENTS];
             "lastEditorEmail":"SFResultsUiT.instr@gmail.tmt",
-            "lastEditedAt":"2016-03-02 11:59 PM UTC",
+            "lastEditedAt":"2026-03-01 11:59 PM UTC",
```
Comment (wkurniawan07) [inline src/test/resources/data/StudentFeedbackResultsPageUiTest.json]: Why is this shifted forward by one day?

Comment (wkurniawan07) [inline src/test/resources/data/StudentFeedbackResultsPageUiTest.json]: I realized there are a lot of occurrences like this. Can you undo it, but still change the year to 2026?

Comment (tarun14110) [inline src/test/resources/data/StudentFeedbackResultsPageUiTest.json]: @wkurniawan07 
We are getting failed tests on 2nd march. So, I had shifted dates from 2nd march to 1st march , so that test cases doesn't fail on that day.
The problem arises when I shifted all dates with 2016 to 2026, past dates of 2016 also moved to 2026.
So, do I need to only revert back only dates or year also of passed dates of 2016 ?

**Comment by thyageshm** [comment]
@tarun14110, @wkurniawan07 lets try and complete this by this release please.

File: src/test/resources/data/StudentFeedbackResultsPageUiTest.json (hunk 1 above)
Comment (wkurniawan07) [inline src/test/resources/data/StudentFeedbackResultsPageUiTest.json]: @tarun14110 Actually, see if you can shift the dates forward instead (e.g 2014 or 2015), but keep it sensible (i.e start time should be before end time, editedAt must be after createdAt, etc.). This way you can see which ones definitely need to be in the future, and for those dates, shift it backward 10 years like what you've done now.
Also, try not to touch the dates or the months to keep the difference minimum.

**Comment by tarun14110** [comment]
@wkurniawan07 ready for your review again. reverted shift of dates.

File: src/test/resources/pages/unregisteredStudentFeedbackResultsPageOpen.html [hunk 2]
```diff
@@ -109,7 +109,7 @@
                   <li class="list-group-item list-group-item-warning" id="responseCommentRow-${comment.id}">
                     <div id="commentBar-${comment.id}">
                       <span class="text-muted">
-                        From: SFResultsUiT.instr@gmail.tmt [Tue, 01 Mar 2016, 11:59 PM UTC] (last edited by SFResultsUiT.instr@gmail.tmt at Wed, 02 Mar 2016, 11:59 PM UTC)
+                        From: SFResultsUiT.instr@gmail.tmt [Sun, 01 Mar 2026, 11:59 PM UTC]
```
Comment (wkurniawan07) [inline src/test/resources/pages/unregisteredStudentFeedbackResultsPageOpen.html]: Have you re-run the tests again? I don't think the last edited date there should be removed.

Comment (tarun14110) [inline src/test/resources/pages/unregisteredStudentFeedbackResultsPageOpen.html]: actually, My test passes on removing last edit. And if I add last edit, test cases fails. There may be something wrong with my setup.
I will reverting back last edit.

**Comment by tarun14110** [comment]
@wkurniawan07 have a look.

File: src/test/resources/pages/unregisteredStudentFeedbackResultsPageOpen.html [hunk 3]
```diff
@@ -109,7 +109,7 @@
                   <li class="list-group-item list-group-item-warning" id="responseCommentRow-${comment.id}">
                     <div id="commentBar-${comment.id}">
                       <span class="text-muted">
-                        From: SFResultsUiT.instr@gmail.tmt [Tue, 01 Mar 2016, 11:59 PM UTC] (last edited by SFResultsUiT.instr@gmail.tmt at Wed, 02 Mar 2016, 11:59 PM UTC)
+                        From: SFResultsUiT.instr@gmail.tmt [Sun, 01 Mar 2026, 11:59 PM UTC] (last edited by SFResultsUiT.instr@gmail.tmt at Wed, 02 Mar 2016, 11:59 PM UTC)
```
Comment (wkurniawan07) [inline src/test/resources/pages/unregisteredStudentFeedbackResultsPageOpen.html]: Shouldn't the last edited at date change here, and in many other places as well?

Comment (tarun14110) [inline src/test/resources/pages/unregisteredStudentFeedbackResultsPageOpen.html]: yeah, its because My repo is not containing this last edited part. Therefore was not able to update that. Will update this in some time.

**Comment by tarun14110** [comment]
@wkurniawan07 I am really sorry for these unnecessary troubles. I should ran the test before sending to you, but my tests are failing randomly, trying to figure out the reason.
Updated , but there were many failed cases. It is difficult for me to figure out which is due to these changes and which are randomly failing.

**Comment by wkurniawan07** [comment]
The changes should be fine as long as all tests can pass, I trust that it is the case.
Can you tell which tests are randomly failing? Is it the ones you mentioned in #4913?

**Comment by thyageshm** [comment]
@wkurniawan07 I believe this one as well.

**Comment by wkurniawan07** [comment]
I just realized that this will break `InstructorCommentsPageUiTest` because it counters the fix I made in #4879. @tarun14110 can you please, please run all tests and make sure everything passes? It's ok if #4879's fix got undone, as long as the tests can be maintained for an extended period of time.

Alternatively, you can fix in another manner: shift the dates forward instead (e.g 2014 or 2015), but keep it sensible (i.e start time before end time, editedAt after createdAt, etc.). This way you can see which ones definitely need to be in the future, and for those dates, shift it backward 10 years like what you've done now.

**Comment by wkurniawan07** [comment]
And, on that account, pushing this to the next milestone.

**Comment by tarun14110** [comment]
updated , have a look. Not able to get green on InstructorCommentsPageUi Because of #4952. But I had corrected failures because of this issue.

**Comment by wkurniawan07** [comment]
OK since you've decided to undo #4879, have a look at the changes in that PR: you should undo everything, in particular the changes I made in `InstructorCommentsPageUiTest`, then fix the HTML files accordingly. Everything else seems to be in good order.

**Comment by tarun14110** [comment]
updated and got green on InstructorCommentsPageUiTest.

File: src/test/resources/pages/instructorFeedbackResultsFilteredByNoSection.html [hunk 4]
```diff
@@ -560,7 +560,7 @@
         </span>
       </div>
     </div>
-    <div class="panel-collapse collapse in" id="panelBodyCollapse-7" style="height: auto;">
+    <div class="panel-collapse collapsing" id="panelBodyCollapse-7" style="height: 50px;">
```
Comment (wkurniawan07) [inline src/test/resources/pages/instructorFeedbackResultsFilteredByNoSection.html]: This shouldn't be part of the change. Anyway, occurrences of this was fixed in the latest `master` branch so you need to sync, and undo changes like this.

**Comment by wkurniawan07** [comment]
One more issue to resolve and it should be all good. Thanks for your perseverance :+1:

File: src/test/resources/pages/instructorFeedbackResultsFilteredByNoSection.html (hunk 4 above)
Comment (tarun14110) [inline src/test/resources/pages/instructorFeedbackResultsFilteredByNoSection.html]: Updated

**Comment by tarun14110** [comment]
Thanks :) You helped a lot.

File: src/test/resources/pages/instructorFeedbackResultsFilteredBySectionB.html [hunk 5]
```diff
@@ -605,7 +605,7 @@
         </span>
       </div>
     </div>
-    <div class="panel-collapse collapse in" id="panelBodyCollapse-9" style="height: auto;">
+    <div class="panel-collapse collapsing" id="panelBodyCollapse-9" style="height: 50px;">
```
Comment (wkurniawan07) [inline src/test/resources/pages/instructorFeedbackResultsFilteredBySectionB.html]: One more item missed.

**Comment by wkurniawan07** [comment]
As far as I can see, only that one needs correction, the rest is good to go.

File: src/test/resources/pages/instructorCommentsPageAddFrc.html [hunk 6]
```diff
@@ -1774,10 +1774,14 @@ <h4 class="col-sm-9 text-color-primary">
                   <tr>
                     <td>
                       <ul class="list-group comments" id="responseCommentTable-1-1-1">
-                        <li class="list-group-item list-group-item-warning giver_display-by-you status_display-private" id="responseCommentRow-1-1-1-1">
+                        <li class="list-group-item list-group-item-warning giver_display-by-you status_display-public" id="responseCommentRow-1-1-1-1">
```
Comment (damithc) [inline src/test/resources/pages/instructorCommentsPageAddFrc.html]: What's the reason for this change?

Comment (wkurniawan07) [inline src/test/resources/pages/instructorCommentsPageAddFrc.html]: Undoing what I did in #4879 as per discussion.

Comment (damithc) [inline src/test/resources/pages/instructorCommentsPageAddFrc.html]: Got it.

File: src/test/resources/pages/instructorFeedbackResultsFilteredBySectionB.html (hunk 5 above)
Comment (tarun14110) [inline src/test/resources/pages/instructorFeedbackResultsFilteredBySectionB.html]: done

**Comment by wkurniawan07** [comment]
One last change is needed: on line 48 of `instructorFeedbackResultsSortQuestionSearch.html`, change `Sat, 30 Apr 2016` to `Thu, 30 Apr 2026`. I can understand why you missed this, #4948 was blocking you from reaching it.
Everything else is fine, I've run all the tests and managed to get green (after the above is fixed).

**Comment by tarun14110** [comment]
Done
